server/resources/user/sessionController.js

```diff
@@ -48,6 +48,56 @@ exports.login = async (req, res, next) => {
   }
 }
 
+exports.mobileLogin = async (req, res, next) => {
+  /**
+   * NOTES: eventually we want to implement something like https://www.npmjs.com/package/react-native-cookies
+   * so that we can use cookie session-management directly on mobile.
+   * for the time being, though, since the header is proving somewhat inaccessible on mobile,
+   * we're going to use this separate login endpoint that returns it as json
+   */
+  if(req.body.username == undefined) {
+    res.send({ success: false, message: "No username present." });
+  } else {
+    req.body.username = req.body.username.toLowerCase();
+    passport.authenticate('local', {session: true},  async (err, passportUser) => {
+      if(err) {
+        console.log("ERR", err)
+        res.send({ success:false, message: "Error authenticating user." });
+      } else if(!passportUser) {
+        res.send({ success:false, message: "Matching user not found." });
+      } else {
+        // User is authenticted. Now get actual user data from the db and log them in
+        const user = await User.findById(passportUser._id)
+        if(!user) {
+          res.send({ success: false, message: "Error logging user in." });
+        } else {
+          req.logIn(user, err => {
+            if(err) { return next(err);}
+
```

Comment:
Not sure why it's `return next(err)` as opposed to `res.send({success: false, message: err})`